src/main/java/nextstep/security/httpsecurity/HttpSecurityConfiguration.java

```diff
@@ -0,0 +1,36 @@
+package nextstep.security.httpsecurity;
+
+import nextstep.security.config.Customizer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Configuration
+public class HttpSecurityConfiguration {
+
+    private static final String BEAN_NAME_PREFIX = "nextstep.security.httpsecurity.HttpSecurityConfiguration.";
+
+    private static final String HTTPSECURITY_BEAN_NAME = BEAN_NAME_PREFIX + "httpSecurity";
+
+    @Autowired
+    private ApplicationContext context;
```

Comment:
```suggestion
    private final ApplicationContext context;

    public HttpSecurityConfiguration(ApplicationContext context) {
        this.context = context;
    }
```
요렇게 표현하는 것이 조금 더 객체의 불변성을 보장하면서 안정적으로 사용할 수 있을 것 같아요 :)